code/modules/projectiles/guns/automagical.dm

```diff
@@ -0,0 +1,70 @@
+//True automagic gun modes. Dakka dakka.
```

Comment:
Since this is really more of an addition to guns, rather than a new type of gun, this should probably be in the default gun file? That'd also probably fix your .dme commit.

Reply:
It's in it's own file mostly because i'd rather not clutter up gun.dm with helper code / objects that aren't directly related to the gun subtype or the fire() procs.

Reply:
I may move it later. My reasoning is this: If it can be used by any gun, which seems to be the case, putting it in a separate file, especially one named like that, will baffle future coders trying to look for it.

Reply:
Feel free to move it later, then. There was previously a block of pseudo code there which was the original code given to me by Mport, which, if you want to see: https://pastebin.com/zf1x7XXD